api/src/main/java/com/redhat/ipaas/api/IPaasEntity.java

```diff
@@ -15,9 +15,15 @@
  */
 package com.redhat.ipaas.api;
 
-public interface IPaasEntity {
+public abstract class IPaasEntity {
 
-	public String getId();
-	public void setId(String id);
+	abstract public String getId();
+	abstract public void setId(String id);
+	
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
There are loads of other things that you want to check for before you put stuff in the cache. ID is just one case. How about e.g. name? What you're looking for is uniqueness on a (number of) field(s) in the cache (ala unique key in a DB table).